packages/dapp-toolkit/src/modules/wallet-request/wallet-request.ts

```diff
@@ -625,30 +480,27 @@ export const WalletRequestModule = (input: {
   const destroy = () => {
     stateModule.destroy()
     requestItemModule.destroy()
+    requestResolverModule.destroy()
     input.providers.transports?.forEach((transport) => transport.destroy())
 
     subscriptions.unsubscribe()
   }
 
   return {
-    sendRequest: (input: { isConnect: boolean; oneTime: boolean }) => {
-      const result = sendRequest({
+    sendRequest: (input: { isConnect: boolean; oneTime: boolean }) =>
+      sendRequest({
         isConnect: input.isConnect,
         oneTime: input.oneTime,
         dataRequestState: dataRequestStateModule.getState(),
       })
-
-      if (connectResponseCallback)
-        result
-          .map((result) => {
-            connectResponseCallback!(ok(result))
-          })
-          .mapErr((error) => {
-            connectResponseCallback!(err(error))
-          })
-
-      return result
-    },
+        .andThen((response) => {
+          if (connectResponseCallback) connectResponseCallback!(ok(response))
```

Comment:
`  if (connectResponseCallback) connectResponseCallback!`

doesn't TS understand that and you need t assert with `!`?